tools/codegen/api/dispatcher.py

```diff
@@ -40,27 +41,25 @@ def returns_type(rs: Sequence[Return]) -> CType:
     # At present, there is no difference. But there could be!
     return cpp.returns_type(rs)
 
-def argument(
-    a: Union[Argument, TensorOptionsArguments, SelfArgument]
-) -> List[Binding]:
-    if isinstance(a, Argument):
-        return [Binding(
-            nctype=argument_type(a, binds=a.name),
-            name=a.name,
-            argument=a,
-        )]
-    elif isinstance(a, SelfArgument):
-        return argument(a.argument)
-    elif isinstance(a, TensorOptionsArguments):
-        return argument(a.dtype) + argument(a.layout) + argument(a.device) + argument(a.pin_memory)
-    else:
-        assert_never(a)
+def jit_arguments(func: FunctionSchema) -> List[Argument]:
```

Comment:
mostly re-jiggered this so that the xla codegen could pattern match on the Argument types, and have them match 1-to-1 with the arguments in each xla kernel (which follow the dispatcher API)

Reply:
just curious, why the "jit" in "jit_arguments"?

Reply:
Mostly
(1) This function returns a list of `Argument` objects which model the jit schema; the only aspect of this function that's dispatcher-specific is that it unpacks `SelfArgument` and `TensorOptionsArguments`.
(2) `def arguments` is already taken and I can't overload in python 😛 if you have a better name in mind feel free.